File: frontend/src/types/news.ts
export type SentimentLabel = "Positive" | "Neutral" | "Negative";

export interface NewsItem {
  id: string;
  title: string;
  source: string;
  summary: string;
  url: string;
  publishedAt: string;
  sentimentLabel: SentimentLabel;
  sentimentScore: number;
}

File: src/main/java/com/citicup/entity/News.java
package com.citicup.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "news")
@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class News {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String source;

    @Column(columnDefinition = "TEXT")
    private String summary;

    private String url;

    private LocalDateTime publishedAt;

    private String sentimentLabel;

    private Double sentimentScore;
}